profile/src/main/java/rdx/works/profile/datastore/EncryptedPreferencesManager.kt

```diff
@@ -153,6 +153,21 @@ class EncryptedPreferencesManager @Inject constructor(
         putString(permanentPreferences, P2P_LINKS_PREFERENCES_KEY, p2pLinkListJson, KeySpec.Profile())
     }
 
+    suspend fun saveP2PLinksWalletPrivateKey(privateKeyHex: String) {
```

Comment:
Maybe this can be replaced with `ExactlyNBytes`? If we know the length of the key.

Reply:
The reason for this is the same as the other comment about the types around the shared preferences code. I see the preferences manager as an IO component that doesn't need to know about any types that it can handle explicitly (primitives in this case).